src/query/core/buildMiddleware/cacheLifecycle.ts

```diff
@@ -254,16 +252,17 @@ export const build: SubMiddlewareBuilder = ({
       const cacheEntryRemoved = new Promise<void>((resolve) => {
         lifecycle.cacheEntryRemoved = resolve
       })
-      const cacheDataLoaded = toOptionalPromise(
-        Promise.race([
-          new Promise<void>((resolve) => {
-            lifecycle.valueResolved = resolve
-          }),
-          cacheEntryRemoved.then(() => {
-            throw neverResolvedError
-          }),
-        ])
-      )
+      const cacheDataLoaded = Promise.race([
+        new Promise<void>((resolve) => {
+          lifecycle.valueResolved = resolve
+        }),
+        cacheEntryRemoved.then(() => {
+          throw neverResolvedError
+        }),
+      ])
+      // prevent uncaught promise rejections from happening.
+      // if the original promise is used in any way, that will create a new promise that will throw again
```

Comment:
I made a quick example to demonstrate this behavior in isolation: https://codesandbox.io/s/interactive-promise-behavior-yv2te?file=/src/index.js